### PR TITLE
Get api/costumes

### LINKED ISSUE #38: Convert GET /api/costumes to Express route
- [x] Create `api/costumes.ts`
- [x] Create an Express router to handle requests to /api/costumes
- [x] Add your costume router to your API router
- [x] Handle GET /api/costumes requests

### PR DESCRIPTION
**Pull request by dyazdani**
## Description
Creating endpoint for getting all costumes under the /api route.

## Related Issue
Closes #38 

## Checklist
- [x] Created tests which fail without the change (if possible)
- [x] Confirmed all tests are passing

## How Has This Been Tested?
Tested that endpoint retrieved all costume data, whether there was just one or if there were more costume entries. Also tested manually with Postman.
Screen Shot 2023-09-27 at 14 02 54
Screen Shot 2023-09-27 at 13 59 21
Screen Shot 2023-09-27 at 13 15 06

### CODE REVIEW

File: __tests__/endpoints/getCostumes.test.js
Comment (paloobi) [inline __tests__/endpoints/getCostumes.test.js:1]: Ah I see why you were asking about setup. Is there a way to write the tests themselves in TypeScript?

In my experience, tests are usually near the code they are testing, and named based on the name of the file they are testinf

So I'd expect suggest this test would be in `server/api/__tests__/costumes.test.ts`

Comment (dyazdani) [reply __tests__/endpoints/getCostumes.test.js:1]: Ah, good point about .ts extension. I have an issue I think to reorganize the tests so that the folders are near the files they're testing.

File: server/api/costumes.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// @ts-ignore
```
Comment (paloobi) [inline server/api/costumes.ts]: Why is this here? We shouldn't need to `ts-ignore` the express import, it is strongly typed

File: server/api/costumes.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import express from "express";
+
+const { getAllCostumes, getPool } = require("../db");
```
Comment (paloobi) [inline server/api/costumes.ts:3]: You may need to ts-ignore this line tho since the DB files aren't typed yet

File: server/api/costumes.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const pool = getPool();
```
Comment (paloobi) [inline server/api/costumes.ts:5]: Oh may have some performance issues in the long-run, because it's creating a new pool for every single request

I filed an issue in the backlog for refactoring this later: #52

File: __tests__/endpoints/getCostumes.test.js
Comment (AnthonyPinto) [reply __tests__/endpoints/getCostumes.test.js:1]: For the purpose of this PR, it is worth putting the new tests in the right location. The pre-existing tests can be handled separately

**Review by AnthonyPinto — CHANGES_REQUESTED** [review]
Requesting changes for relocating the tests and using typescript.

File: server/api/costumes.ts (hunk 1 above)
Comment (dyazdani) [reply server/api/costumes.ts]: Put there by mistake

File: server/api/costumes.ts (hunk 2 above)
Comment (dyazdani) [reply server/api/costumes.ts:3]: TS seems fine without the ts-ignore

File: __tests__/endpoints/getCostumes.test.js
Comment (dyazdani) [reply __tests__/endpoints/getCostumes.test.js:1]: @paloobi requested I leave the `__tests__` folder where it is for now, but reorganize the file structure within it. That's done.

**Review by AnthonyPinto — APPROVED** [review]
Basically LGTM, but I have added a backlog task to either convert the jest tests to use typescript, or definitely decide not to and remove the modules that support that from the project: https://github.com/dyazdani/costume-shop-db/issues/57
